server/bin/pbench-server-prep-shim-002.py

```diff
@@ -11,97 +11,66 @@
 import os
 import sys
 import glob
-import shutil
-import selinux
-import tempfile
 
+from logging import Logger
 from pathlib import Path
 
 from pbench.common.exceptions import BadConfig
 from pbench.common.logger import get_pbench_logger
-from pbench.common.utils import md5sum
 from pbench.server import PbenchServerConfig
-from pbench.server.report import Report
-from pbench.server.utils import quarantine
-from pbench.server.database.models.tracker import Dataset, States, DatasetError
-from pbench.server.database import init_db
+from pbench.cli.agent.commands.results import MoveResults
+from pbench.agent.results import FileUploadError
 
 
 _NAME_ = "pbench-server-prep-shim-002"
 
 
 class Results:
-    def __init__(
-        self, nstatus="", ntotal=0, ntbs=0, nquarantined=0, ndups=0, nerrs=0,
-    ):
+    """Results: Class to collect status details about tar ball
+
+        Args -
+            nstatus -- appends string status of tarball
+            ntotal -- total number of tarballs processed
+            ntbs -- number of tarballs successfully processed
+            nerrs -- number of tarballs that failed
```

Comment:
The parameters should be described in the same order as the formal parameter declarations in `__init__`. Much easier to read that way! (I also would have described the parameters under `__init__` rather than at the class level, as that's where they're used.)

Reply:
Ohh, I will make that change.

Reply:
According to this blog, it has described parameters at both class level as well as inside `__init__` https://realpython.com/documenting-python-code/

Reply:
Hmm, that blog has some interesting advice (though we already have a tradition of not following PEP 257 style); but in the section on class docstrings, that's not quite what it says:

> Class docstrings should contain the following information:
>
> 1. A brief summary of its purpose and behavior
> 1. Any public methods, along with a brief description
> 1. Any class properties (attributes)
> 1. Anything related to the interface for subclassers, if the class is intended to be subclassed
>
> The __class constructor parameters should be documented within the __init__ class method docstring__. Individual methods should be documented using their individual docstrings. 

I haven't been listing class public methods in the class docstring, and I'm uncertain how I feel about that: documentation isn't automatically checked for consistency with code, and it's easy for it to get stale. In general, anything that one can gather straightforwardly by reading the code is better left out of the docstring. (For example, that's why PEP 257, unlike the Google docstring style, omits parameter types in favor of machine-readable type hints.)

The bit about "Anything related to the interface for subclassers" however is great advice I ought to follow in my Elasticsearch query hierarchy, because I've got two (and adding a third) "abstract" methods in the base class assumed to be overridden by the subclass: that _would_ be good to add in detail to the class docstring rather than just in the abstract method docstring.

Reply:
I saw an example and there they have explained the arguments in class, as well as, __init__ method. I might have got carried away with that example ;P.